pkg/trie/triedb/proof/iterator.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package proof
+
+type Iterator[T any] struct {
```

Comment:
Im wondering if this iterator should be located somewhere else in the code. 

Its generic so can be used anywhere, so maybe can have it in a more general place

Reply:
+1 but since the `pkg/trie` aims to be a standalone package I would rather to keep it here